### PR TITLE
Template option

### PR DESCRIPTION
**Pull request by jepke**
Needs more to finish it up but made the start.

### CODE REVIEW

**Comment by bbdoc** [comment]
You need an `if isset` around your new block as well as default to 1 if not set (maybe an hidden input) as I think lots of people won't need it. 

Then you also need to update modal so people can update it as well.

**Comment by jepke** [comment]
That's why i added WIP title tag

**Comment by bbdoc** [comment]
yep no problem, just mentionning so you don't forget it ;-)

btw I see in your current dev you only set $allowed_templates in config, but some people might have different templates for raids/quests/pokemons... we might need different config options for each.

File: modal/eggs_modal.php [hunk 1]
```diff
@@ -64,7 +64,25 @@ class="form-control text-center">
...
+            <div class="col-sm-12 my-1">
+                <div class="btn-group btn-group-toggle" data-toggle="buttons">
+                    <div class="input-group">
+                        <div class="input-group-prepend">
+                            <div class="input-group-text">Template</div>
+                        </div>
+                    </div>';
+                    foreach ( $allowed_templates as $key => $name ) {
+                        echo '<label class="btn btn-secondary">';
+                        echo '<input type="radio" name="template" id="' . $key . '" value="' . $key . '" ' . (($key == $row['template']) ? 'checked' : '') . '>';
```
Comment (jepke) [inline modal/eggs_modal.php:79]: @bbdoc  hope you don't mind that i used a slightly different approach on the checked tag on a button. using ternary line allowed me to use less code.

**Comment by bbdoc** [comment]
Hi,

What's the status of this PR ? Could you please rebase as there have been quite some commits in between. I guess it's also needed for invasions tracking.

My main comment about the PR is that you only have one config for `$allowed_templates` but what if people want to use different templates for pokemons / raids / quests / invasions ? Shouldn't you have one config option per type ?

Did you manage to get feedback on the styling part ?

Thanks

**Comment by jepke** [comment]
> Hi,
> 
> What's the status of this PR ? Could you please rebase as there have been quite some commits in between. I guess it's also needed for invasions tracking.
> 
> My main comment about the PR is that you only have one config for `$allowed_templates` but what if people want to use different templates for pokemons / raids / quests / invasions ? Shouldn't you have one config option per type ?
> 
> Did you manage to get feedback on the styling part ?
> 
> Thanks

It's all split up now. Styling is still a thing imo very minor.

**Comment by bbdoc** [comment]
Is this ready to be merged ? I didn't get any feedback from testing yet.

**Comment by jepke** [comment]
I've been using it since I build it no issues

**Pull request merged**